server/neptune/temporalworker/server.go

```diff
@@ -160,6 +166,11 @@ func (s Server) Start() error {
 		}
 	}()
 
+	// Start job output handler listener
+	go func() {
+		s.JobOutputHandler.Handle()
+	}()
```

Comment:
How does this shut down when there is a sigterm? unless I'm missing something, that global channel doesn't ever get closed and this function just blocks forever?  Can we handle this properly? 

Also what is this waitgroup even doing here? we're supposed to have one per go routine and then wait until they all complete before officially shutting down.  Can you add this for the relevant go routines?